modules/ROOT/pages/as2/as2-connector-upgrade-migrate-6-0.adoc

```diff
@@ -0,0 +1,90 @@
+
+* Certain attribute object fields have been renamed and deprecated fields removed. These are outlined in the Upgrade Steps.
+
+== Requirements and Limitations
+
+You should be aware of some requirements and limitations before you upgrade to the latest version.
+
+=== Requirements
+
+Ensure that you meet the following requirements before you migrate.
+
+* Mule 4.3.0 or later
+
+* Java 8
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade to AS2 version 6.0.0:
+
+. Download the AS2 connector 6.0.0 from Exchange and place appropriate source/operation in the flow that is being upgraded.
+
+=== Changes for AS2 Listener and AS2 MDN Listener
+
+There are no changes to the way AS2 Listener and AS2 MDN Listener is used or configured.
+
+=== Changes for Send with Sync MDN and Send with Async MDN operations
+
+For `Send with Sync MDN` and `Send with Async MDN` operations:
+
+* An attribute object of type `com.mulesoft.connector.as2.api.AS2MdnAttributes` has been changed for v6.0.0. The field `com.mulesoft.connector.as2.api.AS2SendAttributes.as2MdnAttributes` is of type `com.mulesoft.connector.as2.api.AS2MdnAttributes`).
+
+* The MultiMap `dispositionContent` has been converted into an object called `DispositionNotification` splits the information of the disposition-notification part of the mdn into parameters. The `DispositionNotification` can be found in the `DispositionNotificationReport` object.
+
+Amend mule flows that are using these attributes are accordingly.
+
+=== Changes in AS2 MDN attributes
+[%header%autowidth.spread]
+|===
+|v5.x Field Name| v6.0.0 Field Name | v5.x Type | v6.0.0 Type
+|dispositionContent |dispositionContent | MultiMap<String, String> | DispositionNotificationReport
+|mdnMic | DispositionNotificationReport.DispositionNotification.receivedContentMic | String | String
+|mdnMicAlg | DispositionNotificationReport.DispositionNotification.receivedContentMicAlgorithm | HashAlgorithm | HashAlgorithm
+|mdnProcessed | DispositionNotificationReport.DispositionNotification.dispositionType | boolean | String
+|originalAS2MessageId |  DispositionNotificationReport.DispositionNotification.originalMessageID | String | String
+|reportText | Field Removed | String | Field Removed
+|===
+
+=== Verify the Upgrade
+
+After you perform the previous steps, follow these steps to complete the upgrade:
+
+. Verify that the connector is working by making sure there are no exceptions on startup and test it against your AS2 partner.
+
+. Notify your Partners of any new URL (port) configurations that corresponds to the HTTP listener you are using.
+
+== Troubleshooting
+
+If there are problems with caching the parameters and caching the metadata, try restarting Studio.
+
+== Revert the Upgrade
+
+If it is necessary to revert to the previous version of as2MdnAttributes Connector, change the `as2-connector` dependency version in the project's `pom.xml` file to the previous version.
```

Comment:
```suggestion
If you must revert the connector, change the `as2-connector` dependency version in the project's `pom.xml` file to the previous version.
```

Reply:
Consider giving an example:

If you must revert the connector (for example, if this happens), change the...."